include/net/wifi_credentials.h

```diff
@@ -0,0 +1,193 @@
+ * @param[in] password_buf_len		length of password_buf
+ * @param[out] flags			flags
+ * @return 0		Success.
+ * @return -ENOENT	No network with this SSID was found.
+ * @return -EINVAL	A required buffer was NULL.
+ * @return -EPROTO	The network with this SSID is not a personal network.
+ */
+int wifi_credentials_get_by_ssid_personal(
+	const char *ssid,
+	size_t ssid_len,
+	enum wifi_security_type *type,
+	uint8_t *bssid_buf,
+	size_t bssid_buf_len,
+	char *password_buf,
+	size_t password_buf_len,
+	uint32_t *flags
+);
+
+/**
+ * @brief Set credentials for given SSID.
+ * 
+ * @param[in] ssid		SSID to look for
+ * @param[in] ssid_len		length of SSID
+ * @param[in] type		WiFi security type
+ * @param[in] bssid		BSSID (may be NULL)
+ * @param bssid_len		length of BSSID buffer (either 0 or WIFI_MAC_ADDR_LEN)
+ * @param password		password
+ * @param password_len		length of password
+ * @param flags			flags
+ * @return 0			Success. Credentials are stored in persistent storage.
+ * @return -EINVAL		A required buffer was NULL or security type is not supported.
+ * @return -ENOTSUP		Security type is not supported.
+ * @return -ENOBUFS		All slots are already taken.
+ */
+int wifi_credentials_set_personal(
+	const char *ssid,
+	size_t ssid_len,
+	enum wifi_security_type type,
+	const uint8_t *bssid,
+	size_t bssid_len,
+	const char *password,
+	size_t password_len,
+	uint32_t flags
+);
+
+/**
+ * @brief Get credentials for given SSID (raw buffer).
+ *        On success, the buffer will contain one of the following structs:
+ *        `wifi_credentials_header`, `wifi_credentials_psk` or `wifi_credentials_sae`.
+ *
+ * @param[in] ssid		SSID to look for
+ * @param[in] ssid_len		length of SSID
+ * @param[out] buf		buffer to store the struct
+ * @return 0			Success. Results are stored as NULL-terminated strings.
+ * @return -ENOENT		No network with this SSID was found.
+ * @return -EINVAL		A required buffer was NULL or too small.
+ * @return -EPROTO		The network with this SSID is not a personal network.
+ */
+int wifi_credentials_get_by_ssid_personal_raw(const char *ssid, size_t ssid_len,
+					      struct wifi_credentials_personal *buf);
+
+/**
+ * @brief Set credentials for given SSID (raw buffer).
+ *        The buffer has to contain one of the following structs: `wifi_credentials_header`,
+ *        `wifi_credentials_psk` or `wifi_credentials_sae`.
+ *
+ * @param[in] ssid		SSID
+ * @param[in] ssid_len		length of SSID
+ * @param[in] buf		buffer to load the struct from
+ * @return 0			Success.
+ * @return -ENOENT		No network with this SSID was found.
+ * @return -EINVAL		A required buffer was NULL or incorrect size.
+ * @return -EOVERFLOW		All slots are already taken.
+ */
+int wifi_credentials_set_personal_raw(const char *ssid, size_t ssid_len,
+				      const struct wifi_credentials_personal *buf);
+
+/**
+ * @brief Delete credentials for given SSID.
+ *
+ * @param[in] ssid		SSID to look for
+ * @param[in] ssid_len		length of SSID
```

Comment:
Nit: A blank line between parameters and return values makes it a bit easier to read. Same between other types of comments, brief, note etc

Reply:
done